securedrop_client/api_jobs/downloads.py

```diff
@@ -4,21 +4,72 @@
 import os
 import sdclientapi
 import shutil
+from tempfile import NamedTemporaryFile
+from typing import Any, Union, Type, Tuple
 
 from sdclientapi import API
 from sqlalchemy.orm.session import Session
-from typing import Any, Union, Type, Tuple
 
-from securedrop_client import storage
 from securedrop_client.api_jobs.base import ApiJob
 from securedrop_client.crypto import GpgHelper, CryptoError
-from securedrop_client.db import File, Message
-
+from securedrop_client.db import File, Message, Reply
+from securedrop_client.storage import mark_message_as_downloaded, mark_file_as_downloaded, \
+    set_object_decryption_status_with_content
 
 logger = logging.getLogger(__name__)
 
 
-class DownloadSubmissionJob(ApiJob):
+class MessageDownloadJob(ApiJob):
```

Comment:
is the plan still to inherit from a new common object to reduce code duplication with `FileDownloadJob`?

Reply:
Yup! That's the plan: `MessageDownloadJob`, `FileDownloadJob`, and `ReplyDownloadJob` will all inherit from a `DownloadJob` class. I tried to keep the `_decrypt_file` method generic enough so that we can move it into `DownloadJob`. There should be a follow-up PR for this.

I just wanted to keep code changes small here since deleting the message_sync thread felt like a big change.

Reply:
filed #406 for this